mantidimaging/core/operations/rotate_stack/rotate_stack.py

```diff
@@ -96,15 +95,94 @@ def _update_angle(angle, dropdown):
             angle.setEnabled(False)
 
 
-def _rotate_image_inplace(data, angle=None):
-    data[:, :] = rotate(data[:, :], angle)
+def _get_cardinal_angle(value) -> int:
+    """
+    Calculates the angle based on the input value, by adding 45 to the value modulo 360,
+    dividing it by 90, multiplying by 90, and then subtracting 90 if the value modulo 360 modulo
+    90 is equal to 45, otherwise subtracting 0 to get the cardinal angle the aspect ratio should be
+    rotated by.
+
+    param: value: angle of rotation
+    return: angle: cardinal angle of rotation
+    """
+    angle = ((value % 360 + 45) // 90 * 90 - (90 if (value % 360) % 90 == 45 else 0))
+    return angle % 360
+
+
+def _compute_rotation(data, angle, progress) -> np.ndarray:
+    """
+    Handle rotation of image slice by angle. This includes rotation of aspect
+    ratio and in place rotation depending on angle.
+
+    param array_index: index of array in array_list
+    param data: current image data array
+    param angle: angle of rotation
+    return: new_data: rotated image data array
+    """
+    angle += 360 if angle < 0 else 0
+    if angle == _get_cardinal_angle(angle):
+        data.shared_array = _cardinal_rotation_per_slice(data, angle, progress)
+    else:
+        data.shared_array = _cardinal_rotation_per_slice(data, _get_cardinal_angle(angle), progress)
+        data.shared_array = _inplace_rotation(data.data.shape[0], data, angle, _get_cardinal_angle(angle), progress)
```

Comment:
Could we do the subtraction here instead of in the function, i.e. pass `angle - _get_cardinal_angle(angle)`.